internal/controller/drplacementcontrol_controller.go

```diff
@@ -898,6 +844,166 @@ func getPlacementOrPlacementRule(
 	return usrPlacement, nil
 }
 
+func (r *DRPlacementControlReconciler) updateAndSetOwner(
+	ctx context.Context,
+	drpc *rmn.DRPlacementControl,
+	usrPlacement client.Object,
+	log logr.Logger,
+) (bool, error) {
+	// Step 1: Prepare modifications to both objects
+	drpcModified, placementModified, err := r.prepareObjectModifications(drpc, usrPlacement, log)
+	if err != nil {
+		return false, err
+	}
+
+	// Step 2: Apply updates to both objects if needed
+	return r.applyUpdates(ctx, drpc, usrPlacement, drpcModified, placementModified, log)
+}
+
+// prepareObjectModifications prepares all in-memory modifications to both DRPC and placement objects
+func (r *DRPlacementControlReconciler) prepareObjectModifications(
+	drpc *rmn.DRPlacementControl,
+	usrPlacement client.Object,
+	log logr.Logger,
+) (bool, bool, error) {
+	var drpcModified, placementModified bool
+
+	// Annotate the placement object
+	mod, err := r.annotateObject(drpc, usrPlacement, log)
+	if err != nil {
+		return false, false, err
+	}
+	placementModified = placementModified || mod
+
+	// Modify DRPC and placement via metadata updates
+	modDRPC, modPlacement, err := r.updateObjectMetadata(drpc, usrPlacement)
```

Comment:
This should really be 2 functions, one to update the placement and another for the drpc.